homeassistant/components/totalconnect/binary_sensor.py

```diff
@@ -0,0 +1,101 @@
+"""Interfaces with TotalConnect sensors."""
+import logging
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_DOOR,
+    DEVICE_CLASS_GAS,
+    DEVICE_CLASS_SMOKE,
+    BinarySensorDevice,
+)
+
+from . import DOMAIN as TOTALCONNECT_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up a sensor for a TotalConnect device."""
+    if discovery_info is None:
+        return
+
+    sensors = []
+
+    client_locations = hass.data[TOTALCONNECT_DOMAIN].client.locations
+
+    for location_id in client_locations:
+        for zone in client_locations[location_id].zones:
+            sensors.append(
+                TotalConnectBinarySensor(zone, location_id, client_locations)
+            )
+    add_entities(sensors)
+
+
+class TotalConnectBinarySensor(BinarySensorDevice):
+    """Represent an TotalConnect zone."""
+
+    def __init__(self, zone_id, location_id, locations):
+        """Initialize the TotalConnect status."""
+        self._zone_id = zone_id
+        self._location_id = location_id
+        self._zone = locations[location_id].zones[zone_id]
+        self._name = self._zone.description
```

Comment:
How long will the name of the binary_sensor be?

Reply:
I can't find any documentation on a limit in the Total Connect API other than "it's a string."  The names are generally entered by hand at the physical alarm panel, so I don't think we'll find any that are really long.  Some of my names are cut off at 32 characters.  That may be a limitation of my alarm panel, which is an older model, but I don't know for sure.

Reply:
Okay thanks for providing more information about the description.